Formula/elftoolchain.rb

```diff
@@ -0,0 +1,150 @@
+class Elftoolchain < Formula
+  desc "Implementation of compilation tools for the ELF object format"
+  homepage "https://sourceforge.net/p/elftoolchain/wiki/Home/"
+  url "https://downloads.sourceforge.net/project/elftoolchain/Sources/elftoolchain-0.7.1/elftoolchain-0.7.1.tar.bz2"
+  sha256 "44f14591fcf21294387215dd7562f3fb4bec2f42f476cf32420a6bbabb2bd2b5"
+  license all_of: ["BSD-2-Clause", "BSD-3-Clause"]
+  head "https://svn.code.sf.net/p/elftoolchain/code/trunk"
+
+  depends_on "bmake" => :build
+  depends_on "subversion" => :build # TODO: remove in the next release
+  depends_on "libarchive" # needs headers
+
+  uses_from_macos "bison" => :build
+  uses_from_macos "flex" => :build
+  uses_from_macos "m4" => :build
+
+  on_macos do
+    # TODO: add conflicts after merge
+    # conflicts_with "libelf", because: "both install `libelf`"
+    # conflicts_with "dwarfutils", because: "both install `libdwarf`"
+
+    # libpe_dos.c:122:6: error: implicit declaration of function 'htole32' is invalid in C99
+    patch :p0 do
+      url "https://raw.githubusercontent.com/macports/macports-ports/195adb123b397d5ae850d91105f668aff83e8161/devel/elftoolchain/files/patch-byteorder-macros.diff"
+      sha256 "c50570ede5ea2885f63ce513b5061069d1e385abce5c4628819adf24efa647ba"
+    end
+  end
+
+  on_linux do
+    keg_only "it conflicts with binutils and elfutils" # more popular implementations
+  end
+
+  # Symlink binaries that aren't provided by macOS and don't have link conflicts.
+  # - Provided by macOS: ar c++filt ld nm ranlib size strings strip
+  # - Formula conflicts: mcs (mono)
+  def macos_linked_binaries
+    %w[addr2line brandelf elfcopy elfdump findtextrel readelf]
+  end
+
+  def install
+    # Work around to support Apple Silicon in current release.
+    # TODO: Remove in the next release.
+    if build.stable? && Hardware::CPU.arm?
+      ENV.append "CFLAGS", "-DLIBELF_ARCH=EM_AARCH64 -DLIBELF_BYTEORDER=ELFDATA2LSB -DLIBELF_CLASS=ELFCLASS64"
+    end
+
+    # Help find `libarchive` outside default Homebrew prefix
+    ENV.append "CFLAGS", "-I#{Formula["libarchive"].opt_include}"
+    ENV.append "LDFLAGS", "-L#{Formula["libarchive"].opt_lib}"
+
+    # Prevent bmake from running chown
+    inreplace Dir["mk/elftoolchain.{inc,lib}.mk"], "-o ${BINOWN} -g ${BINGRP}", ""
+
+    # install: /usr/local/Cellar/elftoolchain/0.7.1/include/elfdefinitions.h: No such file or directory
+    include.mkpath
+
+    # use libexec on macOS to avoid shadowing commands, which break dependent builds
+    args = %W[
```

Comment:
Is there a `PROGRAM_PREFIX` argument we can use? That seems to make more sense since the binaries that ship with this formula are meant to work on ELF files, so they're not exactly equivalent to the system-provided ones (which is the case for your typical keg-only/unlinked binaries).

Reply:
I didn't see anything that we can easily use.

MacPorts manually does this (i.e. mv cmd to `elftc-#{cmd}` and symlink into `libexec/elftoolchain/#{cmd}`) and also uses subdirectories (e.g. `lib/elftoolchain`, `include/elftoolchain`, ...) to avoid conflicts https://github.com/macports/macports-ports/blob/master/devel/elftoolchain/Portfile#L69-L78

The above would allow us to remove all conflicts/keg-only while having main files installed into linked directories.

Another option is to install binaries under `libexec/...` and just symlink to `bin/SOMEPREFIX-#{cmd}` (e.g. either `elftc-ar` like MacPorts or full name like `elftoolchain-ar`).

Reply:
Fine with doing any of that.